src/pages/release-codes/forms/ReleaseCodeForm.js

```diff
@@ -1,86 +1,70 @@
 import toast from 'react-hot-toast'
 import { RequestsContext } from 'src/providers/RequestsContext'
 import { useInvalidate } from 'src/hooks/resource'
 import { ResourceIds } from 'src/resources/ResourceIds'
-
-// ** Custom Imports
 import CustomTextField from 'src/components/Inputs/CustomTextField'
-import CustomTextArea from 'src/components/Inputs/CustomTextArea'
-
 import { DocumentReleaseRepository } from 'src/repositories/DocumentReleaseRepository'
 import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
 import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { ControlContext } from 'src/providers/ControlContext'
 
 export default function ReleaseCodeForm({ labels, maxAccess, recordId }) {
-  const [isLoading, setIsLoading] = useState(false)
-  const [editMode, setEditMode] = useState(!!recordId)
-
-  const [initialValues, setInitialData] = useState({
-    recordId: null,
-    name: '',
-    reference: ''
-  })
-
+  const { platformLabels } = useContext(ControlContext)
   const { getRequest, postRequest } = useContext(RequestsContext)
 
-  //const editMode = !!recordId
+  const editMode = !!recordId
 
   const invalidate = useInvalidate({
     endpointId: DocumentReleaseRepository.ReleaseCode.page
   })
 
-  const formik = useFormik({
-    initialValues,
+  const { formik } = useForm({
+    initialValues: {
+      recordId: recordId || null,
+      reference: '',
+      name: ''
+    },
     enableReinitialize: true,
     validateOnChange: true,
     validationSchema: yup.object({
-      reference: yup.string().required(' '),
-      name: yup.string().required(' ')
+      reference: yup.string().required(),
+      name: yup.string().required()
     }),
     onSubmit: async obj => {
-      const recordId = obj.recordId
-
-      const response = await postRequest({
+      await postRequest({
         extension: DocumentReleaseRepository.ReleaseCode.set,
         record: JSON.stringify(obj)
       })
+        .then(response => {
+          if (!obj.recordId) {
+            toast.success(platformLabels.Added)
+            formik.setFieldValue('recordId', response.recordId)
+          } else toast.success(platformLabels.Edited)
 
-      if (!recordId) {
-        toast.success('Record Added Successfully')
-        setInitialData({
-          ...obj, // Spread the existing properties
-          recordId: response.recordId // Update only the recordId field
+          invalidate()
         })
-      } else toast.success('Record Edited Successfully')
-      setEditMode(true)
-
-      invalidate()
+        .catch(error => {})
     }
   })
 
   useEffect(() => {
     ;(async function () {
       try {
         if (recordId) {
-          setIsLoading(true)
-
           const res = await getRequest({
             extension: DocumentReleaseRepository.ReleaseCode.get,
             parameters: `_recordId=${recordId}`
           })
 
-          setInitialData(res.record)
+          formik.setValues(res.record)
```

Comment:
below in return part 
the grid tag directly after grow not necessary please remove

Reply:
this is important for spacing between components in form..
and almost same in all screens